Source/JavaScriptCore/bytecode/ArrayAllocationProfile.cpp

```diff
@@ -53,19 +53,20 @@ void ArrayAllocationProfile::updateProfile()
     // So for now, we update the allocation profile only from the main thread.
     
     ASSERT(!isCompilationThread());
-    JSArray* lastArray = std::exchange(m_lastArray, nullptr);
+    Storage storage = std::exchange(m_storage, Storage(nullptr, m_storage.type()));
+    JSArray* lastArray = storage.pointer();
     if (!lastArray)
         return;
     if (LIKELY(Options::useArrayAllocationProfiling())) {
         // The basic model here is that we will upgrade ourselves to whatever the CoW version of lastArray is except ArrayStorage since we don't have CoW ArrayStorage.
-        IndexingType indexingType = leastUpperBoundOfIndexingTypes(m_currentIndexingType & IndexingTypeMask, lastArray->indexingType());
-        if (isCopyOnWrite(m_currentIndexingType)) {
+        IndexingType indexingType = leastUpperBoundOfIndexingTypes(current().indexingType() & IndexingTypeMask, lastArray->indexingType());
+        if (isCopyOnWrite(current().indexingType())) {
```

Comment:
Probably I think we should read `storage`'s indexingType since `current` can be concurrently updated now.

Reply:
I thought about it, but for some reason I thought it wasn't a problem... we only call `setPointer` concurrently, and since it's only a word, it should be safe, no?  In any case, I can update it.